src/ca/mcgill/cs/jetuml/diagram/Diagram.java

```diff
@@ -354,6 +356,11 @@ public void removeEdge(Edge pEdge)
 	public void placeOnTop(Node pNode)
 	{
 		assert pNode != null;
+		// If pNode is a call node in the sequence diagram, do nothing
+		if( aType == DiagramType.SEQUENCE && pNode.getClass() == CallNode.class )

```

Comment:
`CallNode` are only present in sequence diagrams, so I would recommend eliminating the first (presumed defensive) clause `aType == DiagramType.SEQUENCE` because of the potential confusion if may cause.